content/en/agent/autodiscovery/_index.md

```diff
@@ -30,635 +30,70 @@ Here's a 5-minute video for a bird's eye view of Datadog Agent v6 Autodiscovery
 
-Finally, Autodiscovery can load check templates from places other than disk. Other possible **template sources** include key-value stores like Consul, Kubernetes pod annotations, and Docker label annotations.
+The overall process of Datadog Agent Autodiscovery is:
 
-### Different Execution
+1. **Create and Load Integration template**: When the Agent starts with Autodiscovery enabled, it loads integration templates from all [available template sources][2]; along with the [Autodiscovery container identifiers][3]. Static configuration files aren't suitable for checks that collect data from ever-changing network endpoints like host or ports, so Autodiscovery uses [**Template Variables**][4] for integration template configuration. Those integration template configurations can be loaded into the Agent in 4 main ways:
 
-When the Agent starts with Autodiscovery enabled, it loads check templates from all available template sources&mdash;[not just one or another](#template-source-precedence)&mdash;along with the templates' container identifiers. Unlike in a traditional Agent setup, the Agent doesn't run all checks all the time; it decides which checks to enable by inspecting all containers currently running on the same host as the Agent.
+  * [Using a configuration file mounted within the Agent][5]
+  * [Using Key-Value Store][6]
+  * [Using Kubernetes Annotations][7]
+  * [Using Docker Labels][8]
 
-As the Agent inspects each running container, it checks if the container matches any of the container identifiers from any loaded templates. For each match, the Agent generates a static check configuration by substituting the matching container's IP address and port. Then it enables the check using the static configuration.
-
-The Agent watches for Docker events&mdash;container creation, destruction, starts, and stops&mdash;and enables, disables, and regenerates static check configurations on such events.
+2. **Apply an integration template to a specific container**: Unlike in a traditional Agent setup, the Agent doesn't run all checks all the time; it decides which checks to enable by inspecting all containers running on the same host as the Agent and the corresponding loaded integration templates. The Agent then watches for Kubernetes/Docker events&mdash;container creation, destruction, starts, and stops&mdash;and enables, disables, and regenerates static check configurations on such events. As the Agent inspects each running container, it checks if the container matches any of the [Autodiscovery container identifiers][3] from any loaded integration templates. For each match, the Agent generates a static check configuration by substituting the [Template Variables][9] with the matching container's specific values. Then it enables the check using the static configuration.
 
 ## How to set it up
 
-The Datadog Docker Agent automatically auto-discovers other containers.
+**The Datadog Docker Agent automatically auto-discovers other containers**.
```

Comment:
Do we commonly call the agent container `Datadog Docker Agent` ?

I'm not sure this is visible enough when viewing the page it looks like a subtitle and you want to keep reading the host setup, could we have titles for container setup & host setup ?

Also FWIW this is done via env variables when starting the container:
`KUBERNETES=yes` or `